### PR TITLE
[BKM-17] Create step page

### PR DESCRIPTION
**Pull request by nichakontae**
# PR title
Please use the following format for the PR title: `[BKM-NO.] <title>`  

## Description
Create step page

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Other (please describe):

## Changes
List the key changes made in this pull request.
- [ ] Updated feature `<name>`
- [ ] Fixed bug `<description>`
- [x] Added
> creat step page on frontend
> create step controller
> create step service

- [ ] Other: `<brief-description>`

## Proof of Changes (Attach Screenshots, GIFs, or Videos)
Please attach any relevant screenshots, GIFs, or videos that demonstrate the changes are working as expected (especially for UI-related changes or complex logic).

## Notes for Reviewers (optional)
Highlight areas of concern or questions for reviewers.
- Any dependencies or migrations required?
- Known issues with this PR?
- Anything specific you’d like the reviewer to focus on?

## Related Issues 
If this PR addresses an issue, link it here.
- [Issue #<number>](link-to-issue)
- [PR #<number>](link-to-PR)

## Request Review
If you are ready for the code to be reviewed, please add the label **"ready to review"** and contact the following team members:
- @nichakontae

### CODE REVIEW

File: frontend/src/components/step/eval-type.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                                {isLoadingSubmitEval ? "Submitting..." : "Submit"}
+                            </Button>
+                        </div>
+                    )}
+                    {userEvalId && userSubmission && (
+                        <Dialog>
+                            <DialogTrigger asChild>
+                                <div
+                                    className="w-[450px] cursor-pointer pt-4 flex items-center justify-center overflow-hidden"
+                                    style={{ height: "auto" }}
```
Comment (KNattawat89) [inline frontend/src/components/step/eval-type.tsx:146]: set height ใน className

File: frontend/src/components/step/eval-type.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                            <Loader2 className="animate-spin" />
+                            <p>Please wait, this may take a few minutes as we check your answer</p>
+                        </div>
+                    )}
+                </div>
+            )}
+
+            {type !== "check" && passStatus !== null && passStatus !== undefined && commentStatus !== null && commentStatus !== undefined && (
+                <ResultCard pass={passStatus} comment={commentStatus} />
+            )}
```
Comment (KNattawat89) [inline frontend/src/components/step/eval-type.tsx:205]: อันนี้สร้าง variable มาตัวนึง เพื่อไว้check condition พวกนี้ code จะได้ clean

File: frontend/src/components/ui/avatar.tsx [hunk 3]
```diff
@@ -1,48 +1,31 @@
...
-      "flex h-full w-full items-center justify-center rounded-full bg-muted",
-      className
-    )}
-    {...props}
-  />
+    <AvatarPrimitive.Fallback
+        ref={ref}
+        className={cn("flex h-full w-full items-center justify-center rounded-full bg-gray-200", className)}
+        {...props}
+    />
```
Comment (KNattawat89) [inline frontend/src/components/ui/avatar.tsx:25]: อันนี้เขียนstyleใน className ข้างนอกไหม (หรือลองแล้วไม่ได้?) เผื่ออนาคตมีใช้componentนี้กับที่อื่น ที่background color สีต่างกัน

Comment (nichakontae) [reply frontend/src/components/ui/avatar.tsx:25]: อ่อ อันนี้ลืมลบ แค่ตอนแรกใช้สีเทามันสวยกว่า5555555

File: frontend/src/pages/portal/index.tsx
Comment (nichakontae) [inline frontend/src/pages/portal/index.tsx:1]: don't forget to revert this page, this is only for testing

File: frontend/src/pages/home/index.tsx
Comment (nichakontae) [inline frontend/src/pages/home/index.tsx:1]: don't forget to revert change, this is only for testing

**Pull request closed without merging**